pkg/clustersql/settings.go

```diff
@@ -27,21 +27,23 @@ import (
 	"github.com/dustin/go-humanize"
 )
 
+// ErrInvalidClusterSettingName is returned when the supplied setting name isn't valid.
+var ErrInvalidClusterSettingName = fmt.Errorf("only letters, numbers, underscores, and periods are allowed")
+
 // Cluster names have letters, underscores, and periods. See here for more:
 // https://www.cockroachlabs.com/docs/stable/cluster-settings.html
 var validClusterSettingNameRE = regexp.MustCompile(`^[a-z_.\d]+$`)
 
-//IsValidClusterSettingName func
-func IsValidClusterSettingName(name string) error {
+func validateSettingName(name string) error {
```

Comment:
Why are we returning an err?  Would it make sense to return a boolean?

Reply:
At first, I thought the same thing. Especially when it was `IsValidClusterSettingName`. However, since this is used in both `GetClusterSetting` and `SetClusterSetting` and each would need to return an error in that case. I un-exported this and leveraged the same error in both cases.

I can update to use a boolean if you feel strongly about it.

Reply:
I do not feel strongly about much in life. 🤣